code/iaas/auth-logic/src/main/java/io/cattle/platform/iaas/api/auth/github/ProjectMemberResourceManager.java

```diff
@@ -0,0 +1,123 @@
+import io.cattle.platform.iaas.api.auth.dao.AuthDao;
+import io.cattle.platform.iaas.api.auth.github.resource.Member;
+import io.cattle.platform.json.JsonMapper;
+import io.cattle.platform.object.ObjectManager;
+import io.cattle.platform.object.process.ObjectProcessManager;
+import io.github.ibuildthecloud.gdapi.context.ApiContext;
+import io.github.ibuildthecloud.gdapi.exception.ClientVisibleException;
+import io.github.ibuildthecloud.gdapi.factory.SchemaFactory;
+import io.github.ibuildthecloud.gdapi.model.ListOptions;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.util.ResponseCodes;
+
+import java.util.*;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class ProjectMemberResourceManager extends AbstractObjectResourceManager {
+
+    @Inject
+    JsonMapper jsonMapper;
+    @Inject
+    AuthDao authDao;
+    @Inject
+    GenericResourceDao resourceDao;
+    @Inject
+    ObjectManager objectManager;
+    @Inject
+    ObjectProcessManager objectProcessManager;
+
+    private List<ExternalIdHandler> externalIdHandlers;
+
+    @Override
+    protected Object removeFromStore(String type, String id, Object obj, ApiRequest request) {
+        return null;
+    }
+
+    @Override
+    protected Object listInternal(SchemaFactory schemaFactory, String type, Map<Object, Object> criteria, ListOptions options) {
+        Policy policy = (Policy) ApiContext.getContext().getPolicy();
+        String id = (String) criteria.get("id");
+        if (StringUtils.isNotEmpty(id)) {
+            ProjectMember projectMember = authDao.getProjectMember(Long.valueOf(id));
+            if (!authDao.hasAccessToProject(projectMember.getProjectId(), policy.getAccountId(),
+                    policy.isOption(Policy.AUTHORIZED_FOR_ALL_ACCOUNTS), policy.getExternalIds())) {
+                throw new ClientVisibleException(ResponseCodes.NOT_FOUND);
+            }
+            policy.grantObjectAccess(projectMember);
+            return Arrays.asList(projectMember);
+        }
+        id = (String) criteria.get("projectId");
+        if (StringUtils.isNotEmpty(id)) {
+            List<? extends ProjectMember> members = authDao.getActiveProjectMembers(Long.valueOf(id));
+            for (ProjectMember member: members){
+                policy.grantObjectAccess(member);
+            }
+            return members;
+        }
+        List<? extends ProjectMember> members = authDao.getActiveProjectMembers(policy.getAccountId());
+        for (ProjectMember member: members){
+            policy.grantObjectAccess(member);
+        }
+        return members;
+    }
+
+    @Override
+    public String[] getTypes() {
+        return new String[] { "projectMember" };
+    }
+
+    @Override
+    public Class<?>[] getTypeClasses() {
+        return new Class<?>[0];
+    }
+
+    public List<ProjectMember> setMembers(Account project, List<Map<String, String>> members) {
```

Comment:
Since this isn't used in this class but is used in ProjectResourceManager, I think we should move it into the ProjectResourceManager class.  Then you won't even need to inject ProjectMemberResourceManager into the ProjectResourceManager class.  Plus, I think logically it makes more sense too.  You're setting members of a project.